feature-libs/product/multi-dimensional/components/multi-dimensional-container/multi-dimensional-selector/product-multi-dimensional-selector.component.ts

```diff
@@ -0,0 +1,67 @@
+/*
+ * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {
+  ChangeDetectionStrategy,
+  Component,
+  inject,
+  Input,
+  OnChanges,
+  SimpleChanges,
+} from '@angular/core';
+import {
+  Product,
+  ProductScope,
+  ProductService,
+  RoutingService,
+  VariantCategory,
+} from '@spartacus/core';
+import { ProductMultiDimensionalService } from '../../../core/services/product-multi-dimensional.service';
+import { ActivatedRoute } from '@angular/router';
+import { filter, take } from 'rxjs/operators';
+
+@Component({
+  selector: 'cx-variants-multi-dimensional-selector',
+  templateUrl: './product-multi-dimensional-selector.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class ProductMultiDimensionalSelectorComponent implements OnChanges {
+  protected multiDimensionalService: ProductMultiDimensionalService = inject(
+    ProductMultiDimensionalService
+  );
+  protected productService: ProductService = inject(ProductService);
+  protected routingService: RoutingService = inject(RoutingService);
+  protected route: ActivatedRoute = inject(ActivatedRoute);
+
+  @Input()
+  product: Product;
+
+  categories: VariantCategory[] = [];
+
+  ngOnChanges(changes: SimpleChanges) {
+    if (changes.product) {
+      this.categories = this.multiDimensionalService.getVariants(this.product);
+    }
+  }
+
+  changeVariant(code: string | undefined): void {
+    if (code) {
+      this.productService
+        .get(code, ProductScope.MULTIDIMENSIONAL)
```

Comment:
[MEDIUM]
Let's use the scope `LIST` to build URLs, as in other places in our system

Reply:
I'm not sure I understand this. If I use the LIST scope, I'll lose some of the product attributes I need to determine which product to navigate to.
The "variants" library does the same thing.

Reply:
Oh God, I didn't realize it is the component. I thought you referred to the component guard...

Reply:
fixed